Remove internal executorch dependency on torchao.quantization.subclass

andrewor14 opened the pull request:
Summary:
**Summary:** This is a really old quantization API that we
recently removed in torchao (D84842047). No one should be
calling it anymore. For BC, let's just copy the base class
into executorch for now. We should delete this in the future.

**Test Plan:** CI

Differential Revision: D84921134

andrewor14 commented:
@larryliu0820

lucylq commented on examples/models/llama/experimental/subclass.py:
```diff
@@ -129,6 +132,172 @@ def to_float(
     return a * scale.unsqueeze(1)
 
 
+class QuantizedLinearWeightBase(torch.Tensor):
```
Can we tag this as deprecated? 

```
from executorch.exir._warnings import deprecated
.. 
@deprecated("QuantizedLinearWeightBase is deleted from torchao ... ")
class QuantizedLinearWeightBase(torch.Tensor):
```

andrewor14 committed:
Remove internal executorch dependency on torchao.quantization.subclass (#15223)

Summary:
Pull Request resolved: https://github.com/pytorch/executorch/pull/15223

**Summary:** This is a really old quantization API that we
recently removed in torchao (D84842047). No one should be
calling it anymore. For BC, let's just copy the base class
into executorch for now. We should delete this in the future.

**Test Plan:** CI

bypass-github-export-checks

Reviewed By: vkuzo

Differential Revision: D84921134
